src/Venturecraft/Revisionable/RevisionableTrait.php

```diff
@@ -184,14 +184,31 @@ public function postSave()
             $changes_to_record = $this->changedRevisionableFields();
 
             $revisions = array();
-
+            
+            $getJsonData = function ($value) {
+                $jsonData = json_decode($value);
+                if (is_array($jsonData) || is_object($jsonData)) return (array)$jsonData;
+                if (is_array($value) || is_object($value)) return (array)$value;
+                else return $value;
+            };
+            
             foreach ($changes_to_record as $key => $change) {
+                $originalData = $getJsonData(array_get($this->originalData, $key));
```

Comment:
This won't work in Laravel 7+, since the helpers no longer exist by default. For compatibility with more Laravel versions it should be:
```diff
-$originalData = $getJsonData(array_get($this->originalData, $key));
+$originalData = $getJsonData(Arr::get($this->originalData, $key));
```

Reply:
This has nothing to do with the `Array to string conversion` error though, this one's an easy fix 😉

Reply:
Awesome!